### PR TITLE
Fix #3364

### LINKED ISSUE #3364: WARNING in nginx: Received undecodable user/password from nginx: 'aaron@my.domain'/'password�password'
## Environment & Version

- Docker Compose version v2.29.1
- Version: MAILU_VERSION:-2024.06

## Description
After upgrading to 2024.06 some users with % in their password generate an error:

```
WARNING in nginx: Received undecodable user/password from nginx: 'aaron@my.domain'/'password�password'
```
(FYI I've replaced the actual mail domain and password with password and left the special character in i got in the logs as example)

## Replication Steps
Set % in the password which generates special characters.

## Expected behaviour
It should not have issues with % in the password.

### PR DESCRIPTION
**Pull request by nextgens**
## What type of PR?

bug-fix

## What does this PR do?

Fix a bug preventing percent characters from being used in passwords

### Related issue(s)
- closes #3364 

## Prerequisites
Before we can consider review and merge, please make sure the following list is done and checked.
If an entry in not applicable, you can check it or remove it from the list.

- [ ] In case of feature or enhancement: documentation updated accordingly
- [x] Unless it's docs or a minor change: add [changelog](https://mailu.io/master/contributors/workflow.html#changelog) entry file.

### CODE REVIEW

**Comment by nextgens** [comment]
Let's see if the tests pass now.

bors try

**Comment by nextgens** [comment]
bors try

**Comment by nextgens** [comment]
bors retry

**Comment by nextgens** [comment]
Try with characters that can't be expressed in latin1

bors try

**Comment by nextgens** [comment]
The only question remaining is why we unquote() the Client-IP...

**Comment by nextgens** [comment]
I guess it's for ipv6 ; not a problem as long as we quote it too.

**Comment by nextgens** [comment]
bors try

**Review by ghostwheel42 — DISMISSED** [review]

File: core/admin/mailu/internal/views/auth.py [hunk 1]
```diff
@@ -50,14 +49,8 @@ def nginx_authentication():
...
-        password = None
-        try:
-            password = raw_password.encode("iso8859-1").decode("utf8")
-        except:
-            app.logger.warn(f'Received undecodable password for {username} from nginx: {raw_password!r}')
-            utils.limiter.rate_limit_user(username, client_ip, password=None)
-        else:
-            utils.limiter.rate_limit_user(username, client_ip, password=password)
+        password = urllib.parse.unquote(headers.get('Auth-Pass', None))
+        utils.limiter.rate_limit_user(username, client_ip, password=password)
```
[the remark this answers is not in the document]
Comment (nextgens) [reply core/admin/mailu/internal/views/auth.py]: good catch, that's in 78c5d342

File: core/admin/mailu/internal/nginx.py [hunk 2]
```diff
@@ -91,20 +91,14 @@ def handle_authentication(headers):
...
         password = 'invalid'
         try:
-            user_email = raw_user_email.encode("iso8859-1").decode("utf8")
-            password = raw_password.encode("iso8859-1").decode("utf8")
+            user_email = urllib.parse.unquote(headers["Auth-User"])
+            password = urllib.parse.unquote(headers["Auth-Pass"])
             ip = urllib.parse.unquote(headers["Client-Ip"])
         except:
-            app.logger.warn(f'Received undecodable user/password from nginx: {raw_user_email!r}/{raw_password!r}')
+            app.logger.warn(f'Received undecodable user/password from nginx: {headers["Auth-User"]!r}/{headers["Auth-Pass"]!r}')
```
[the remark this answers is not in the document]
Comment (nextgens) [reply core/admin/mailu/internal/nginx.py]: It's arguable; I've removed it in 78c5d342

File: core/admin/mailu/internal/nginx.py [hunk 3]
```diff
@@ -91,20 +91,14 @@ def handle_authentication(headers):
...
-        # we need to manually decode.
-        raw_user_email = urllib.parse.unquote(headers["Auth-User"])
-        raw_password = urllib.parse.unquote(headers["Auth-Pass"])
         user_email = 'invalid'
         password = 'invalid'
         try:
-            user_email = raw_user_email.encode("iso8859-1").decode("utf8")
-            password = raw_password.encode("iso8859-1").decode("utf8")
+            user_email = urllib.parse.unquote(headers["Auth-User"])
+            password = urllib.parse.unquote(headers["Auth-Pass"])
```
[the remark this answers is not in the document]
Comment (nextgens) [reply core/admin/mailu/internal/nginx.py:98]: I don't think that's the case; The code it replaces wouldn't have handled it; Here we will just go to the Expect and log which is fine

[the remark this answers is not in the document]
Comment (ghostwheel42) [reply core/admin/mailu/internal/nginx.py:98]: Okay. it's okay, there's a try/except around.

**Review by ghostwheel42 — APPROVED** [review]
LGTM

**Pull request merged**